### PR TITLE
feature/APPEALS-43597 RC

### PR DESCRIPTION
**Pull request by ThorntonMatthew**
# VA Notify Resiliency
[APPEALS-43597](https://jira.devops.va.gov/browse/APPEALS-43597)

### CODE REVIEW

File: app/jobs/application_job.rb
Comment (mchbidwell) [inline app/jobs/application_job.rb:1]: Do the jobs need to have the aws-sdk-sqs updated as well?

Comment (ThorntonMatthew) [reply app/jobs/application_job.rb:1]: They do, and that gets pulled in by including `aws-sdk` in the Gemfile.

I tried including just `awsk-sdk-sqs` and it caused us to not be able to push/pull messages from our queues... For now I've opted to leave the entire SDK in place as it is in production currently.

File: app/jobs/application_job.rb [hunk 1]
```diff
@@ -18,6 +18,21 @@ def application_attr(app_name)
...
+    # See https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html
+    def delete_sqs_message_before_start?
+      false
+    end
+
+    # For jobs that run multiple times in a short time span, we do not want to continually update
+    # the JobsExecutionTime table. This boolean will help us ignore those jobs
+    def ignore_job_execution_time?
+      false
+    end
```
Comment (noahhansen-gov) [inline app/jobs/application_job.rb:34]: Is there any time in which we would return true for these? If we always return false can we just replace all instances of these methods with `false`

File: app/jobs/ignore_job_execution_time.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    base.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    # Include this module in job classes if you anticipate that the job will run
+    # several times in a short time span. As a result, the Job execution time
+    # does not need to be tracked
+    def ignore_job_execution_time?
+      true
+    end
```
Comment (noahhansen-gov) [inline app/jobs/ignore_job_execution_time.rb:14]: Same comment as above. Why have the method if it only returns one value?

File: app/services/external_api/va_notify_service.rb [hunk 3]
```diff
@@ -33,20 +33,20 @@ class << self
...
-      email_template_id,
-      first_name,
-      docket_number,
-      status = ""
+      participant_id:,
+      notification_id:,
+      email_template_id:,
+      first_name:,
+      docket_number:,
+      status: ""
```
Comment (noahhansen-gov) [inline app/services/external_api/va_notify_service.rb:41]: Might be missing some ruby context but do these just default to `nil`? Should we specify that here?

Comment (noahhansen-gov) [reply app/services/external_api/va_notify_service.rb:41]: Includes all other instances of this

**Review by noahhansen-gov — COMMENTED** [review]
Some small comments!

File: app/jobs/application_job.rb (hunk 1 above)
Comment (ThorntonMatthew) [reply app/jobs/application_job.rb:34]: We override them using these modules:
- [MessageConfigurations::DeleteMessageBeforeStart](https://github.com/department-of-veterans-affairs/caseflow/blob/b0ddb65d55d3b53af14f578173eaad2751c52292/app/jobs/message_configurations/delete_message_before_start.rb#L3)
- [IgnoreJobExecutionTime](https://github.com/department-of-veterans-affairs/caseflow/blob/b0ddb65d55d3b53af14f578173eaad2751c52292/app/jobs/ignore_job_execution_time.rb#L3)

Comment (ThorntonMatthew) [reply app/jobs/application_job.rb:34]: Wherever those are included, then for those jobs those methods will return `true`.

File: app/services/external_api/va_notify_service.rb (hunk 3 above)
Comment (ThorntonMatthew) [reply app/services/external_api/va_notify_service.rb:41]: I think we'd want to throw an error if a caller failed to provide any necessary information. I don't think there are solid default values for these, unfortunately.

**Review by noahhansen-gov — APPROVED** [review]

**Pull request merged**